### ISSUE TITLE
Changes are stuck in Still modifying... state even after changes are applied

### ISSUE THREAD

**Issue by RichardMore** [Telmate/terraform-provider-proxmox#831]
Hi,
I am having a weird issue where any change I apply will be stuck in still modifying.
Example: change disk size = "50G" -> "51G", the plan is correct
VM is stopped as well.

CLI output:
```
  # proxmox_vm_qemu.test1 will be updated in-place
  ~ resource "proxmox_vm_qemu" "test1" {
        id                        = "pve/qemu/201"
        name                      = "test1"
        # (37 unchanged attributes hidden)

      ~ disk {
          ~ size               = "50G" -> "51G"
            # (27 unchanged attributes hidden)
        }

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
proxmox_vm_qemu.test1: Modifying... [id=pve/qemu/201]
proxmox_vm_qemu.test1: Still modifying... [id=pve/qemu/201, 10s elapsed]
[...]
proxmox_vm_qemu.test1: Still modifying... [id=pve/qemu/201, 10m20s elapsed]
^CStopping operation...

Interrupt received.
Please wait for Terraform to exit or data loss may occur.
Gracefully shutting down...

proxmox_vm_qemu.test1: Still modifying... [id=pve/qemu/201, 10m30s elapsed]
[...]
proxmox_vm_qemu.test1: Still modifying... [id=pve/qemu/201, 15m40s elapsed]
^C
Two interrupts received. Exiting immediately. Note that data loss may have occurred.

╷
│ Error: operation canceled
```

And Proxmox shows that the operation stopped OK, duration: 1s

Edit: Proxmox 8.0.3

**Comment by Laurent977** [comment]
Same for me, every time i need to modify any VM configuration

Proxmox 8.0.3

**Comment by EdwardCooke** [comment]
Just ran into this same issue and after a bit of troubleshooting and digging through the code I figured it out.
After it applies the updates it attempts to get the ip address from the api at this line

https://github.com/Telmate/terraform-provider-proxmox/blob/4c0c8fc37ed3f6316ad9392a46a0cf28c4654a11/proxmox/resource_vm_qemu.go#L2142

That api call returns a 500 when the guest agent isn't running. Here's the log message for that
```
check ip result error 500 QEMU guest agent is not running
```

After I turned on the VM and installed and started the `qemu-guest-agent` package, terraform completed.

Here's what I did on my Debian 12 install
```bash
sudo apt update && sudo apt install -y qemu-guest-agent
sudo systemctl enable qemu-guest-agent
sudo systemctl start qemu-guest-agent
```

Kind of a bummer that the vm has to turn on and be running for an update to complete.

I was able to get the logs by adding the following to the `provider` section for `proxmox`
```terraform
provider "proxmox" {
    pm_api_url = "https://***:8006/api2/json"
    pm_debug = true
    pm_log_enable = true
    pm_log_file   = "terraform-plugin-proxmox.log"
    pm_log_levels = {
        _default    = "debug"
    }
}
```
